[KB-217] Supervisor feedback for introduction (#135)
chhoumann/thesis-chemcam: report_thesis/src/sections/introduction.tex

Part of this research is facilitated through interpretation of spectral data gathered by \gls{libs} instruments, which fire a high-powered laser at soil samples to create a plasma.
The emitted light is captured by spectrometers and analyzed using machine learning models to assess the presence and concentration of certain major oxides, informing NASA's understanding of Mars' geology.

However, predicting major oxide compositions from \gls{libs} data still presents significant computational challenges.
These include the high dimensionality and non-linearity of the data, compounded by issues of multicollinearity and matrix effects~\cite{andersonImprovedAccuracyQuantitative2017}.
Such effects can cause the intensity of emission lines from an element to vary independently of that element's concentration, introducing unknown variables that complicate the analysis.
Furthermore, the high cost of data collection often results in small datasets, exacerbating the difficulty of building accurate and robust models.

> Previous work has aimed to improve the prediction of major oxide compositions from LIBS data.

Reviewer: ```suggestion
Previous work has aimed to improve the prediction of major oxide compositions from \gls{libs} data.
```

> Regression techniques and dimensionality reduction with feature selection have been used to enhance the accuracy and interpretability of these models.

Reviewer: > these models

Which models?

Author: Rephrased this

> Tailored approaches have also been developed, where different models are selected based on their performance with specific spectral characteristics and distances~cite{rezaei_dimensionality_reduction, andersonPostlandingMajorElement2022}.

Reviewer: ```suggestion
Tailored approaches have also been developed, where different models are selected based on their performance with specific spectral characteristics and distances~\cite{rezaei_dimensionality_reduction, andersonPostlandingMajorElement2022}.
```

I don't think 'distances' make sense to readers here. Not readily apparent.

Author: Removed 'distances' since 'spectral characteristics' also encapsulates the distance (since the characteristics change based on the distance)

Moreover, models incorporating physical principles have demonstrated improved accuracy by handling residuals that traditional models fail to explain~\cite{song_DF-K-ELM}.

> However, predicting oxide compositions remains challenging due to the complex, nonlinear nature of LIBS data.

Reviewer: ```suggestion
However, predicting oxide compositions remains challenging due to the complex, nonlinear nature of \gls{libs} data.
```

This underscores the need for continued research into more adaptive and robust machine learning strategies to tackle these issues effectively.

> This thesis aims to improve the prediction of major oxide compositions from LIBS data by building on the foundation established in~\citet{p9_paper} in order to enhance model accuracy and robustness.

Reviewer: `\gls{libs}`

There's something about this sentence structure that isn't right to me.

We aim to improve predictions of elemental concentrations from LIBS data → by building on our previous work → to improve the model accuracy and robustness.

I'd rather focus on the goals in one sentence, and then mention what we're building on in another.
Get to the point when stating a goal.

Like:

> This thesis aims to improve the model accuracy and robustness in predicting major oxide compositions from \gls{libs} data.

And then say something like `we build on the foundation laid in ...`.

Perhaps we should also include the part about 'tackling the challenges in predicting... from LIBS data?'

Author: Rephrased this

> We define accuracy as the ability of a model to predict the composition of major oxides in Martian geological samples, while robustness refers to the stability of these predictions across different samples and oxides.

Reviewer: I deleted this from the intro in the problem definition PR IIRC. It's part of the problem definition now.

Reviewer: Need to merge in?

Author: Done merged and removed this from the introduction.

> To achieve these objectives, this research systematically explores a range of promising machine learning models and preprocessing techniques, identified through an extensive literature review and guided by a curiosity to explore unconventional approaches.

Reviewer: Wonder if it would make sense to put the "we build upon (p9)" here? As part of how we achieve the objectives.

Author: Rephrased

> Specifically, we implemented an experimental framework using an automated hyperparameter optimization tool to determine the most effective combinations of preprocessing methods and models for each major oxide analyzed.

Reviewer: The term "experimental framework" might imply that the framework itself is experimental. While kind of true, I don't think that's what we're trying to communicate here.

What about:
```suggestion
Specifically, we designed and implemented a framework for experimental analysis using an automated hyperparameter optimization tool to determine the most effective combinations of preprocessing methods and models for each target of analysis.
```

Reviewer: That was actually intentional, to just refer to some generic regression target. The method should work for most predictive systems, so specifying it as only working with oxides is reductive of our efforts IMO.

Should probably have said that haha. Thoughts?

Author: Maybe "for each regression target" then? To clarify.

Reviewer: Yep good

Author: ```suggestion
Specifically, we designed and implemented a framework for experimental analysis using an automated hyperparameter optimization tool to determine the most effective combinations of preprocessing methods and models for each regression target.
```

> We began by evaluating various preprocessing techniques to understand their impact on model performance, selecting those that demonstrated the highest impact on improving the performance of each model.

Reviewer: Is this right? IIRC, we started by using only norm3. Only after the second round of reading/experimenting did we start varying the preprocessing steps.

Author: Should now be fixed, I rephrased it.

> Following the preprocessing, the most promising models underwent an optimization process, allowing us to precisely tune both model configurations and their respective hyperparameters on a per-oxide basis, ensuring optimal performance tailored to the specific data characteristics of each oxide.

Reviewer: Model configuration sounds analogous to hyperparameters here. I think you are referring to which configuration of oxides are present in an oxide's ensemble, which is getting a bit ahead of ourselves?

But that doesn't sound right - please clarify!

Once the best configurations were identified, a stacking ensemble method was employed to create a meta learner for each oxide, significantly enhancing prediction accuracy and robustness beyond the capabilities of individual models.
Through extensive experiments on \gls{libs} data, we systematically assessed and demonstrated the superior performance of our approach compared to existing methods, focusing on significant improvements in prediction accuracy and robustness.

Our key contributions are as follows:
\begin{itemize}

>     \item We develop a novel machine learning pipeline that effectively handles the challenges of LIBS data to accurately predict major oxide compositions in Martian soil samples.

Reviewer: ```suggestion
    \item We develop a novel machine learning pipeline that effectively handles the challenges of \gls{libs} data to accurately predict major oxide compositions in Martian soil samples.
```

Reviewer: If we say `effectively handles`, we probably also need to prove it. How? We can show better accuracy and robustness, but the other one is more challenging. Especially given our (lack of) knowledge in non-CS fields here...

Author: ```
\item We develop novel machine learning pipeline that demonstrates improved accuracy and robustness in predicting major oxide compositions in LIBS calibration target.
``

    \item We have developed a novel optimization approach and tool for tuning and evaluating machine learning models along with preprocessing techniques, providing a systematic and efficient method for selecting the best configuration.

>     \item Our investigations provides a methodological framework that advances the approach to model selection and tuning for high-dimensional, multicollinear geochemical data analysis.

Reviewer: What exactly are you referring to here?
Do we know that it advances anything when we didn't have a baseline?

Author: I assume part of the experiments section is comparing our results to previous work and showing that we have improved upon it? Which is what this is referring to.

Reviewer: We'll be comparing the results of our experiments & final pipeline, not the process of model selection, right?

Here, we say that we provide a framework for model selection that's better ('advances' the approach). What is it better than?
It is our results that we believe show improvements. And while we think that our approach to model selection is good, we don't have any basis for comparison - that hasn't been a focus for us this semester.

Author: ¿Any suggestions on how to phrase it then? I see your point but unsure how else to phrase it, or whether we should keep it.

Reviewer: ```suggestion
```

>     \item Using this framework, we have identified and tuned the most promising models and preprocessing techniques from the literature for predicting major oxide compositions in Martian soil samples.

Reviewer: This is something we did. It's not a contribution. It's an action.

Author: Do we remove it then? Or should it somehow be rephrased?

Reviewer: Remove IMO

>     \item By outperforming existing methods, our approach has established new benchmarks for accuracy and robustness in \gls{libs} data analysis, providing a new standard for future research.

Reviewer: ```suggestion
    \item By outperforming existing methods, our approach has established new benchmarks for accuracy and robustness in \gls{libs} data analysis.
```

Such effects can cause the intensity of emission lines from an element to vary independently of that element's concentration, introducing unknown variables that complicate the analysis.
Furthermore, the high cost of data collection often results in small datasets, exacerbating the difficulty of building accurate and robust models.

Previous work has aimed to improve the prediction of major oxide compositions from \gls{libs} data by using regression techniques and dimensionality reduction with feature selection.
These methods have been used to enhance both the accuracy and interpretability of the prediction models.

> Tailored approaches have also been developed, where different models are selected based on their performance with specific spectral characteristics~cite{rezaei_dimensionality_reduction, andersonPostlandingMajorElement2022}.

Reviewer: Cite is broken?
```suggestion
Tailored approaches have also been developed, where different models are selected based on their performance with specific spectral characteristics~\cite{rezaei_dimensionality_reduction, andersonPostlandingMajorElement2022}.
```
